### PR TITLE
Add list handler to ResourceDefaultVersion, remove TypeVersion and Ty…

### PR DESCRIPTION
**Pull request by ugudip**
*Description of changes:*
Create ListHandler for ResourceDefaultVersion, removed TypeVersion and TypeDefaultVersion as they are not in use

Re-ran the list contract tests to test the newly added ListHandler

```
handler_delete.py::contract_delete_list PASSED 
handler_create.py::contract_create_list_success PASSED    
handler_create.py::contract_update_list_success PASSED    
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-cloudformation-resourcedefaultversion/aws-cloudformation-resourcedefaultversion.json [hunk 1]
```diff
@@ -63,6 +63,11 @@
       "permissions": [
         ""
       ]
+    },
+    "list": {
+      "permissions": [
+        "cloudformation:ListTypes"
```
Comment (cathlh) [inline aws-cloudformation-resourcedefaultversion/aws-cloudformation-resourcedefaultversion.json:69]: This should be ListTypeVersions right?

File: aws-cloudformation-resourcedefaultversion/resource-role.yaml [hunk 2]
```diff
@@ -24,6 +24,7 @@ Resources:
               - Effect: Allow
                 Action:
                 - "cloudformation:DescribeType"
+                - "cloudformation:ListTypes"
```
Comment (cathlh) [inline aws-cloudformation-resourcedefaultversion/resource-role.yaml:27]: ListTypeVersions

**Review by cathlh — CHANGES_REQUESTED** [review]

File: aws-cloudformation-resourcedefaultversion/aws-cloudformation-resourcedefaultversion.json (hunk 1 above)
Comment (ugudip) [reply aws-cloudformation-resourcedefaultversion/aws-cloudformation-resourcedefaultversion.json]: yea, my bad, fixed it.

File: aws-cloudformation-resourcedefaultversion/src/main/java/software/amazon/cloudformation/resourcedefaultversion/Translator.java [hunk 3]
```diff
@@ -57,7 +57,7 @@ static ResourceModel translateFromReadResponse(@NonNull final DescribeTypeRespon
     static List<ResourceModel> translateFromListResponse(@NonNull final ListTypeVersionsResponse awsResponse) {
         return streamOfOrEmpty(awsResponse.typeVersionSummaries())
                 .map(typeSummary -> ResourceModel.builder()
-                        .arn(typeSummary.arn())
+                        .arn(typeSummary.arn().substring(0, typeSummary.arn().lastIndexOf("/")))
```
Comment (cathlh) [inline aws-cloudformation-resourcedefaultversion/src/main/java/software/amazon/cloudformation/resourcedefaultversion/Translator.java:60]: does this exclude the version part of the arn? we want to return the versioned arn here i thought

Comment (ugudip) [reply aws-cloudformation-resourcedefaultversion/src/main/java/software/amazon/cloudformation/resourcedefaultversion/Translator.java:60]: No, the arn for ResourceDefaultVersion is without the version. This is done to make it unique per resource.

Comment (cathlh) [reply aws-cloudformation-resourcedefaultversion/src/main/java/software/amazon/cloudformation/resourcedefaultversion/Translator.java:60]: got it, thanks for clarifying

**Review by cathlh — APPROVED** [review]

File: aws-cloudformation-typedefaultversion/src/main/java/software/amazon/cloudformation/typedefaultversion/CallbackContext.java [hunk 4]
```diff
@@ -1,10 +0,0 @@
-package software.amazon.cloudformation.typedefaultversion;
-
```
Comment (rajdivya) [inline aws-cloudformation-typedefaultversion/src/main/java/software/amazon/cloudformation/typedefaultversion/CallbackContext.java:2]: Why this and a bunch of other files being deleted?

Comment (ugudip) [reply aws-cloudformation-typedefaultversion/src/main/java/software/amazon/cloudformation/typedefaultversion/CallbackContext.java:2]: TypeDefaultVersion and TypeVersion were added before to deprecate existing Modules* and Resource* but later we decided to move on with separate ones, hence I am deleting Type* files as they are not used anywhere.

**Review by rajdivya — APPROVED** [review]

**Pull request merged**